models/organization.go

```diff
@@ -0,0 +1,218 @@
+		"name = ?",
+		name,
+	).First(&org).Error; err != nil {
+		return nil, err
+	}
+	return org, nil
+}
+
+// GetOrgUsers is used toreturn the registered users an organization has
+func (om *OrgManager) GetOrgUsers(name string) ([]string, error) {
+	org, err := om.FindByName(name)
+	if err != nil {
+		return nil, err
+	}
+	return org.RegisteredUsers, nil
+}
+
+// IncreaseAccountBalance increases the amount owed by this account
+func (om *OrgManager) IncreaseAccountBalance(name string, amount float64) error {
+	org, err := om.FindByName(name)
+	if err != nil {
+		return err
+	}
+	if org.AccountBalance+amount < org.AccountBalance {
+		return errors.New("account balance overflow error")
+	}
+	org.AccountBalance = org.AccountBalance + amount
+	return om.DB.Model(org).Update("account_balance", org.AccountBalance).Error
+}
+
+// DecreaseAccountBalance decreases the amount owed by this account
+func (om *OrgManager) DecreaseAccountBalance(name string, amount float64) error {
+	org, err := om.FindByName(name)
+	if err != nil {
+		return err
+	}
+	if org.AccountBalance-amount > org.AccountBalance {
+		return errors.New("account balance overflow error")
+	}
+	org.AccountBalance = org.AccountBalance - amount
+	return om.DB.Model(org).Update("account_balance", org.AccountBalance).Error
+}
+
+// GetTotalStorageUsed returns the total storage in bytes consumed
+// by the organization.
+func (om *OrgManager) GetTotalStorageUsed(name string) (uint64, error) {
+	org, err := om.FindByName(name)
+	if err != nil {
+		return 0, err
+	}
+	var total uint64
+	for _, user := range org.RegisteredUsers {
+		usage, err := NewUsageManager(om.DB).
+			FindByUserName(user)
+		if err != nil {
+			return 0, err
+		}
+		if total+usage.CurrentDataUsedBytes < total {
+			return 0, errors.New("overflow error")
+		}
+		total = total + usage.CurrentDataUsedBytes
+	}
+	return total, nil
+}
+
+// BillingReport contains a summary
+// of an organizations entire active
+// user base in the last 30 days along with
+// the USD value currently owned by the account
+type BillingReport struct {
+	Name  string        `json:"name"`
+	Items []BillingItem `json:"items"`
+	// amount owed in USD
+	AmountDue float64 `json:"amount_due"`
+	// the unix (nano) timestamp the report was finalized at
+	Time int64 `json:"time"`
+}
+
+// BillingItem is an individual user's
+// billing history
+type BillingItem struct {
+	User    string   `json:"user"`
+	Uploads []Upload `json:"uploads"`
+}
+
+// GenerateBillingReport is used to generate a billing report object for an
+// organization's entire user base in the last 30 days. Care must be taken so that
+// only the organization owner may interact with this function, and is it returns sensitive information
+func (om *OrgManager) GenerateBillingReport(name string, minTime, maxTime time.Time) (*BillingReport, error) {
+	org, err := om.FindByName(name)
+	if err != nil {
+		return nil, err
+	}
+	report := &BillingReport{Name: name, AmountDue: org.AccountBalance}
```

Comment:
So AccountBalance is actually amount owing. I find this variable name confusing too.

Reply:
What do you think would be better?

Reply:
good idea